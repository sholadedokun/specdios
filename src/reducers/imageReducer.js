import { FETCH_IMAGES } from '../actions/actionTypes';

export default function(state = {}, action) {
  switch(action.type) {
    case FETCH_IMAGES:
        return { ...state, error: '', emailNotification: 'Please Wait...', allImages:action.payload };
    default: return {...state}
  }
}
