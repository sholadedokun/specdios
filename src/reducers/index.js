import { combineReducers } from 'redux';
import { reducer as form } from 'redux-form';
import userReducer from './userReducer';
import inventoryReducer from './mealInventoryReducer';
import settings from './settingsReducer'

const rootReducer = combineReducers({
  form,
  inventory: inventoryReducer,
  user:userReducer,
  settings
});
export default rootReducer;
