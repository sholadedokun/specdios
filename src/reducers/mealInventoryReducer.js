import {
  FETCH_CATEGORIES,
  FETCH_MEAL_TYPES,
  FETCH_NUTRITION,
  FETCH_INGREDIENTS,
  ADD_NEW_MEAL,
  UNAUTH_USER,
  AUTH_ERROR,
  FETCH_OFFERS
} from '../actions/actionTypes';

export default function(state = {}, action) {
  switch(action.type) {
    case FETCH_CATEGORIES:
      return { ...state, error: '', allCategories: action.payload };
    case FETCH_MEAL_TYPES:
      return { ...state, allMealTypes:action.payload };
    case FETCH_NUTRITION:
        return { ...state, error:'', allNutritions:action.payload };
    case FETCH_INGREDIENTS:
        return { ...state, error:'', allIngredients:action.payload };
    case ADD_NEW_MEAL:
      return { ...state, error:'', meal:action.payload };
    case FETCH_OFFERS:
      return { ...state, products:action.payload };
  }
  return state;
}
