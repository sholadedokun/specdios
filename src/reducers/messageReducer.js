import { SENDING, SEND_EMAILS, SEND_EMAILS_ERROR } from '../actions/actionTypes';

export default function(state = {}, action) {
  switch(action.type) {
    case SENDING:
        return { ...state, error: '', emailNotification: 'Please Wait...' };
    case SEND_EMAILS:
        return { ...state, error: '', emailNotification: action.payload };
    case SEND_EMAILS_ERROR:
        console.log(action.payload)
        return { ...state, error: action.payload, emailNotification: '' };
    default: return {...state}
  }
}
