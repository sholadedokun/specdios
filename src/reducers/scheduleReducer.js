import { FETCH_SCHEDULES, FETCH_SCHEDULES_ERROR } from '../actions/actionTypes';
import {schedule} from '../config';
export default function(state = {}, action) {
  switch(action.type) {
    case FETCH_SCHEDULES:
        let allTimeSlot = action.payload.AvailableDates;
        let allSchedules = action.payload.bookedSlot;
        let newPayload=allTimeSlot.reduce((payload, item)=>{
            let data={}
            let allBookedSlots= allSchedules.filter((bookedItem)=>{
                return (bookedItem.date == item)
            })
            //there are sessions already booked on this date
            if(allBookedSlots.length < 2){
                data.date=item
                //reduce all available sessions (allAvail sessions - all booked sessions)
                let availableSessions = schedule.sessionPeriods.reduce((sessPayload, sItem)=>{
                    //filter out all used sessions
                    let allBookedSess= allBookedSlots.filter((bookedItem)=>{
                        return (bookedItem.slot == sItem.from)
                    })
                    if(allBookedSess.length == 0){
                        return sessPayload.concat(sItem)
                    }
                    return sessPayload;
                },[])

                data.AvailSession=availableSessions
                return payload.concat(data);
            }
            else{
                return payload
            }
        },[])
        return { ...state, error: '', allSchedules: newPayload };
    case FETCH_SCHEDULES_ERROR:
        return { ...state, error: action.payload, allSchedules: '' };
    default: return {...state}
  }
}
