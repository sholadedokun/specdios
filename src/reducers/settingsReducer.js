import {
    ADD_NEW_SETTINGS,
    RESET_MEAL_TYPE,
} from '../actions/actionTypes';

export default function(state = {}, action) {
  switch(action.type) {
    case ADD_NEW_SETTINGS:
        return { ...state, error: '', newSettings: action.payload };
    default: return {...state}
  }
}
