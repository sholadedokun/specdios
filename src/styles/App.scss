@import url('https://fonts.googleapis.com/css?family=Lato:400,700');
@import url('https://fonts.googleapis.com/css?family=Raleway:300,300italic,500,500italic');
@import 'font-awesome.min.css';
body{
    margin: 0;
    padding: 0;
    font-family: 'Maven Pro','sans-serif';
    height: 100%;
}

.nop{
    padding: 0;
}
.contentVIew{
    img{
        position: relative;

        cursor: pointer;
        margin: 1em 40%;
        cursor: pointer;
    }
    position: relative;
     background:#191919;
     padding: 3em;
     min-height:100vh;
     color: #eee;
     width:35em;
}
.menu{
    display: flex;
    flex-direction: column;
    position: fixed;
    width:auto;
    min-width: 120px;
    height:100vh;
    background: #222;
    padding:25vh 0 4em !important;
    color:#eee !important;
    z-index: 3;
    box-shadow: 2px 0px 15px 1px #151515;

    img{
        margin:0 auto;
    }
    li{
        padding:0.5em 1em;
        display:block !important;
        cursor: pointer;
        margin:5px 0;
        text-align: center;
    }
    .active{
        color:#fcf693;
    }
}
ul{
    margin: 0;
    padding: 0;
    list-style: none;
}
.menuList{
    margin-top:4em;
}

.logoHead{
    font-family: 'Playfair Display', 'Exo';
    font-weight: bold;
    letter-spacing: 0.1em;
    text-transform: uppercase;
    color: #024b8d;
    font-size: 3em;
}

h3, h4, h5{
    font-weight: 400;
    margin: 10px 0 20px;
}
h1, h2, h3, h4, h5, {
    font-family: 'Lato', 'Raleway';
    font-weight: bold;
}

.logoSubHead{
    font-size: 1em;
    letter-spacing: 0.2em;
    line-height: 0px;
    text-transform: uppercase;
    font-family: 'Exo';
    display: block;
    margin-left: 9.5em
}
.workSlide{
    margin-top: 100px;
    margin-bottom: 20px;
    font-size: 2em;
    display: block;
    height: 1.3em;
    overflow: hidden;
    font-family: 'Exo';
}
button, input[type="submit"]{
    padding: 1em;
    color: #333;
    background: linear-gradient(#fcf693, #ffd271);
    border:none;
    border-radius: 5px;
    display: block;
    outline: none;
}
.selectInput{
  display: inline-block;
  background:#fff;
  border-radius: 4px;
  padding:10px;
  width: 100%;
  margin-bottom: 1em;
  select{
    border:0;
    color:#000;
    width:100%;
    outline:none;
    padding: 10px;
  }
}
.introText{
  position:absolute;
  background: #eee;
  border: 2px solid #fff;
  opacity: 0.9;
  padding: 2.5em;
  h2{
      font-size: 1.3em;
  }
  h3{
    font-size: 1em;
    margin:0
  }
  h4{
      font-size: 0.85em
  }
  .closeButton{
    font-size:2em;
    color:#555;
    position: absolute;
    top: 1em;
    right:1.2em;
    cursor:pointer;
    // box-shadow: 0px 0px 1px 1 #777;
  }
  border-radius: 5px;
    width: 350px;
    margin: 190px 200px;
    button{
      display: block;
      width: auto;
      margin-top:2.5em;
      box-shadow: 0 0 5px 1px #aaa;
    }
}
.imageMover{
     position:fixed;
     height:100%;
     width:100%;

}
.videoContainer{
     position:absolute;
     background: #eee;
     z-index: 2;
     height:100vh;
     width:100vw;
     text-align: center;
     video{
         width: 100%;
         height: 100%;
     }
     .skip{
         font-weight: bold;
         color: #333;
         display: inline-block;
         margin-left:  50vw;
         padding:10px 20px;
         background:#ccc;
         cursor: pointer;
         position: absolute;
         bottom: 10em;
         left:0;
         z-index: 2;
         opacity: 0.85;
         border:2px #fff solid
     }

}
.infoSpace{
    position: absolute;
    width:auto;
    height: 100vh;
    left:8vw;
}
.notification{
    color:#a49;
}
.textError{
    color:#a33
}
input, textarea{
    background:#eee;
    color: #555;
    border: 1px solid #ddd;
    display:block;
    width:100%;
    border-radius: 5px;
    padding:0.8em 1em;
    margin: 1em 0;
    outline: none;
}
input:focus{
    border:1px solid #ffd271;
    background: #fff
}
.footerMenu{
  background: linear-gradient(#3a3a3a,#303030);
      width: 100vw;
      box-shadow: 0 -1px 10px #222;
      padding: 10px 20px;
      font-size: .9em;
      position: fixed;
      bottom: 0;
      z-index: 4;
      color: #666;
      text-align: left;
      ul{
        float: right;
        position: relative;
      }
      li {
        color: #999 !important;
        float: left;
        height: 20px;
        margin: 0 5px;
        cursor: pointer;
      }
      li:hover{
        color:#ddd;
      }
}
.route{
    position: relative;
    z-index: 2;
}
a{
  text-decoration: none;
  margin: 0;
  color:#888;
}
a:hover{

  color:#ddd;
}

//animations keyframes
@keyframes rotate{
  0%{
        transform: rotate(0deg);
        opacity: 1;
    }
  25%{
        transform: rotate(90deg);
        opacity: 1;
    }
  50%{
      transform: rotate(90deg);
      opacity: 0;
    }
    51%{
        transform: rotate(0deg);
        opacity: 0;
      }
  75%{
      transform: rotate(0deg);
      opacity: 1;
  }
  100%{
      transform: rotate(0deg);
      opacity: 1;
  }
}

@keyframes promoEnter{
    from{
      transform: translateX(320px);
      opacity: 0;
      z-index: 1;}

    to{
      transform: translateX(0px);
      opacity: 1;
      z-index: 0;}
}
@keyframes promoLeave{
    from{
      transform: translateX(0px);
      opacity: 1;}

    to{
      transform: translateX(320px);
      opacity:0;}
}
@keyframes moveEnter{
    from{
      transform: translateX(-400px);
      opacity: 1;
      z-index: 1;}

    to{
      transform: translateX(0px);
      opacity: 1;
      z-index: 1;}
}
@keyframes moveLeave{
    from{
      transform: translateX(0px);
      opacity: 1;}

    to{
      transform: translateX(-400px);
      opacity:1;}
}

@keyframes picEnter{
    0%{
        transform: translateY(-100vh) ;
        z-index: 0;
        opacity: 0;
    }

    80%{
      transform: translateY(15vh);
      opacity: 0.85;
    }
      100%{
        transform: translateY(0vh);
        opacity: 1;
    }
}
@keyframes picLeave{
    0%{
      transform: translateY(0vh);
      opacity: 1;
      z-index: 0;
    }
    20%{
        transform: translateY(15vh)
    }
    75%{
        opacity: 0
    }

    100%{
      transform: translateY(-100vh) ;
      opacity:0;
      z-index: 0;
    }
}

.animate_xs_delay{
    animation-delay: 500ms !important;
}


.animate_sm_delay{
    animation-delay: 750ms !important;
}


.animate_md_delay{
    animation-delay: 1500ms !important;}

.animate_lg_delay{
    animation-delay: 2000ms !important;}

.animate_xs_duration{
    animation-duration: 500ms !important;}

.animate_sm_duration{
    animation-duration: 750ms !important;}

.animate_md_duration{
    animation-duration: 1500ms !important}

.animate_lg_duration{
    animation-duration: 2000ms !important}

.animate_xlg_duration{
    animation-duration: 3000ms !important}
.rotateClockwise{
    animation: rotate 4s linear infinite }

.rotateAntiClockwise{
    animation:rotate 1s  linear infinite  reverse}
.transform_origin{
    transform-origin: 100% 100%}

.promoAnim-enter{
    opacity: 0;
}
.promoAnim-enter.promoAnim-enter-active{
  animation: promoEnter 1230ms ease-in 0ms forwards}
.promoAnim-leave{
    opacity: 1;
}
.promoAnim-leave.promoAnim-leave-active{
  animation: promoLeave 1250ms ease-in 0ms forwards}

.aboutImages-enter{
  opacity: 0.01;
  z-index: 4}

.aboutImages-enter.aboutImages-enter-active{
  animation: moveEnter 900ms ease-in 0ms forwards}

.aboutImages-leave{
  opacity: 1;
   z-index: 1}

.aboutImages-leave.aboutImages-leave-active{
  animation: moveLeave 1000ms ease-in 0ms forwards
}

.picAnim-enter{
    opacity: 0;
}
.picAnim-enter.picAnim-enter-active{
  animation: picEnter 530ms ease-in 200ms forwards}
.picAnim-leave{
    opacity: 1;
}
.picAnim-leave.picAnim-leave-active{
  animation: picLeave 1250ms ease-in 0ms forwards}

.aboutItem{
    margin: 0;
    h2, span{
        margin: 0 0 0.5em 0 !important}
}
.aboutMovable{
     width: 100% !important;
     .inner{
         display: flex;
         flex-direction: row;
     }
    ul{
        padding: 0;
        margin: 0;
        display: inline-block;
        text-align: left;
        list-style: none;
        -webkit-animation-name: change;
        -webkit-animation-duration: 10s;
        -webkit-animation-iteration-count: infinite;
        animation-name: change;
        animation-duration: 10s;
        animation-iteration-count: infinite;
        li{
            margin:0;
            padding: 0
        }
    }
}
.imageMover{
    position: fixed;
    height: 100vh;
    width: 100vw;
    background: #333;
    overflow: hidden;
    .imageContainer{
        height: 100%;
        position: relative;
        top: 0;
        .imgBack{
            position: absolute;
            display: inline-block;
            width: 100%;
            height: 100%;
        }
    }
}
#warning-message {
     display: none;
     text-align: center;
     padding-top: 30vh;
     height:100vh;
     background: url("/images/mobileBackground.jpg") center no-repeat;
     img{
         display: block;
         margin: 1em auto;
     }
     .iconLg{
         font-size: 4em !important;
     }
     .warning-content{
         background: #fff;
         color:#333;
         padding:20px;
         opacity: 0.95;
         border:1px solid #333;
         border-left: none;
         border-right: none;
     }
     .icon-stack{

         margin: 1em 0 0.5em -1em;
         min-height: 2em;
         position: relative;
         font-size: 2em !important;
         span{
             width:auto;
             position: absolute;
             display: inline-block;
         }
     }
     .icon-stack span:nth-child(1){
         font-size: 2.4em !important;
         margin:-0.25em 0 0 -0.32em;
     }
 }
 @media only screen and (max-width: 800px){
     .introText{
         margin-top: 4em;
         margin-left: 9.5em;
         button{
             margin-top: 1em;
         }
     }
 }
@media screen and (max-width: 700px){
    .menu{
        width:50px;
        min-width: 30px;
        img{
            display: none;
        }
        .menuList{
            margin-top: 0;
        }
    }
    .introText{
        margin-top: 2em;
        margin-left: 4.5em;
        button{
            margin-top: 1em;
        }
    }

}
@media screen and (min-height: 590px){
    @media screen and (max-width: 1024px){
        .imageContainer{
            padding: 5.5em 0;
        }
    }
    @media screen and (max-width: 1380px){
        .imageContainer{
            padding: 8em 0;
        }
    }
}


   @media only screen and (orientation:portrait){
       .App{ display:none; }
       #warning-message { display:block; }
   }
   @media only screen and (orientation:landscape){
       #warning-message { display:none; }
   }
