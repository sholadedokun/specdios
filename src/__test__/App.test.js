import React from 'react';
import { shallow, mount, render} from 'enzyme'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import { createStore, applyMiddleware } from 'redux';
import reduxThunk from 'redux-thunk';


import App from '../components/app';
import Header from '../components/header'
import reducers from '../reducers';


//create a Snapshot for the App Layout
describe('>>>H O M E --- Snapshot',()=>{

    //applying reduxThunk as middleware enabled us to use dispatch from actions
    const createStoreWithMiddleware = applyMiddleware(reduxThunk)(createStore);
    const store = createStoreWithMiddleware(reducers);
    let wrapper

    beforeEach(()=>{
        // store = mockStore(initialState)
        // wrapper = mount( <Provider store={store}><ConnectedHome /></Provider> )
    })
    it('+++capturing Snapshot of App Layout', () => {
        const renderedValue =  renderer.create(<Provider store={store}><App /></Provider>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});

it('renders without crashing', () => {
  let  wrapper = shallow(<App />);
  expect(wrapper.contains(<Header  title="Evrifod" />)).toBe(true);
});
