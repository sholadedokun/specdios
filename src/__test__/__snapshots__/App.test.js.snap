// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Snapshot +++capturing Snapshot of App Layout 1`] = `
<div
  className="App nop container-fluid"
>
  <div
    className="row"
  >
    <div
      className="App-header col-xs-12"
    >
      <div
        className="container"
      >
        <div
          className="companyName"
        >
          Evrifod
        </div>
        <ul
          className="nav"
          role={null}
        >
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            onSelect={null}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            onSelect={null}
          >
            <a
              href="/inventory"
              onClick={[Function]}
            >
              inventory
            </a>
          </li>
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            onSelect={null}
          >
            <a
              href="/pricing"
              onClick={[Function]}
            >
              Pricing
            </a>
          </li>
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            onSelect={null}
          >
            <a
              href="/help"
              onClick={[Function]}
            >
              Help
            </a>
          </li>
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            className="nav-item"
            onSelect={null}
          >
            <a
              className="nav-link"
              href="/signin"
              onClick={[Function]}
            >
              Sign In
            </a>
          </li>
          <li
            active={undefined}
            activeHref={undefined}
            activeKey={undefined}
            className="nav-item"
            onSelect={null}
          >
            <a
              className="nav-link"
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="restaurantHead  row"
      >
        <div
          className="nop col-xs-8"
        >
           
          <span />
           Found
        </div>
        <div
          className="nop col-xs-4"
        >
          Evrifod Space
        </div>
      </div>
    </div>
  </div>
</div>
`;
